ttnn/conv.py

```diff
@@ -0,0 +1,137 @@
+# SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+
+# SPDX-License-Identifier: Apache-2.0
+
+import math
+import pathlib
+from typing import Optional, Tuple, Union, Dict
+
+
+import tt_lib as ttl
+
+from ttnn.tensor import (
+    Shape,
+    Tensor,
+    Device,
+    from_torch,
+    to_torch,
+    to_device,
+    from_device,
+    to_layout,
+    DataType,
+    MemoryConfig,
+    MathFidelity,
+    DRAM_MEMORY_CONFIG,
+    Layout,
+    ROW_MAJOR_LAYOUT,
+    TILE_LAYOUT,
+    TILE_SIZE,
+    has_storage_type_of,
+)
+
+from tt_eager.tt_dnn.op_library.sliding_window_op_infra.tt_py_composite_conv import (
+    TTPyCompositeConv,
+    SlidingWindowOpParams,
+)
+
+
+class Conv2D:
```

Comment:
Can we move all input shapes together? Maybe something like:
- [B, C, H, W]
- [K, R, S]

Reply:
We want to make the arguments similar to the pytorch Conv2D op. The pytorch version of the conv op has arguments for "in_channels/C", "out_channels/K" and the filter parameters are defined by "kernel_size". But it doesn't take the tensor input shape as arguments for building the op. Those args are required for TTNN op. There is a downside of user providing input tensor shape as a list of ints as we would have to specify the order of the dims - NCHW or NHWC. We can avoid this by having it split into 3 args - batch size, input height, input width.